lib/cdc/util/optional_schema_test.go

```diff
@@ -1,29 +1,122 @@
+				"email":      typing.String,
+			},
+		},
+		{
+			body: MySQLDelete,
+			expected: map[string]typing.KindDetails{
+				"id":         typing.Integer,
+				"first_name": typing.String,
+				"last_name":  typing.String,
+				"email":      typing.String,
+			},
+		},
+		{
+			body: PostgresDelete,
+			expected: map[string]typing.KindDetails{
+				"id":         typing.Integer,
+				"first_name": typing.String,
+				"last_name":  typing.String,
+				"email":      typing.String,
+			},
+		},
+		{
+			body: PostgresUpdate,
+			expected: map[string]typing.KindDetails{
+				"id":           typing.Integer,
+				"first_name":   typing.String,
+				"last_name":    typing.String,
+				"email":        typing.String,
+				"boolean_test": typing.Boolean,
+				"bool_test":    typing.Boolean,
+				"bit_test":     typing.Boolean,
+				"numeric_test": {
+					Kind:                   typing.EDecimal.Kind,
+					ExtendedDecimalDetails: decimal.NewDecimal(ptr.ToInt(decimal.PrecisionNotSpecified), decimal.DefaultScale, nil),
+				},
+				"numeric_5": {
+					Kind:                   typing.EDecimal.Kind,
+					ExtendedDecimalDetails: decimal.NewDecimal(ptr.ToInt(5), 0, nil),
+				},
+				"numeric_5_2": {
+					Kind:                   typing.EDecimal.Kind,
+					ExtendedDecimalDetails: decimal.NewDecimal(ptr.ToInt(5), 2, nil),
+				},
+				"numeric_5_6": {
+					Kind:                   typing.EDecimal.Kind,
+					ExtendedDecimalDetails: decimal.NewDecimal(ptr.ToInt(5), 6, nil),
+				},
+				"numeric_5_0": {
+					Kind:                   typing.EDecimal.Kind,
+					ExtendedDecimalDetails: decimal.NewDecimal(ptr.ToInt(5), 0, nil),
+				},
+				"numeric_39_0": {
+					Kind:                   typing.EDecimal.Kind,
+					ExtendedDecimalDetails: decimal.NewDecimal(ptr.ToInt(39), 0, nil),
+				},
+				"numeric_39_2": {
+					Kind:                   typing.EDecimal.Kind,
+					ExtendedDecimalDetails: decimal.NewDecimal(ptr.ToInt(39), 2, nil),
+				},
+				"numeric_39_6": {
+					Kind:                   typing.EDecimal.Kind,
+					ExtendedDecimalDetails: decimal.NewDecimal(ptr.ToInt(39), 6, nil),
+				},
+			},
 		},
 	}
 
-	for _, tc := range tcs {
-		actualData := tc.s.GetOptionalSchema()
-		assert.Equal(t, tc.expected, actualData, tc.name)
+	for idx, tc := range tcs {
+		var schemaEventPayload SchemaEventPayload
+		err := json.Unmarshal([]byte(tc.body), &schemaEventPayload)
+		assert.NoError(t, err, idx)
+
+		actualData := schemaEventPayload.GetOptionalSchema()
+		for actualKey, actualVal := range actualData {
+			testMsg := fmt.Sprintf("key: %s, actualKind: %s, index: %d", actualKey, actualVal.Kind, idx)
+
+			expectedValue, isOk := tc.expected[actualKey]
+			assert.True(t, isOk, testMsg)
+			assert.Equal(t, expectedValue.Kind, actualVal.Kind, testMsg)
+			if expectedValue.ExtendedDecimalDetails != nil || actualVal.ExtendedDecimalDetails != nil {
+				assert.Equal(t, expectedValue.ExtendedDecimalDetails.Scale(), actualVal.ExtendedDecimalDetails.Scale(), testMsg)
```

Comment:
`assert.NotNil(t, actualVal.ExtendedDecimalDetails, testMsg)`

Reply:
And then below:
```
} else {
    assert.Nil(t, actualVal.ExtendedDecimalDetails, testMsg)
}```

Reply:
So that if `ExtendedDecimalDetails` erroneously comes back `nil` for `actualVal` the test fails.

Reply:
I like it